source/common/lua/wrappers.cc

```diff
@@ -23,5 +23,91 @@ int BufferWrapper::luaGetBytes(lua_State* state) {
   return 1;
 }
 
+void MetadataMapWrapper::setValue(lua_State* state, const ProtobufWkt::Value&& value) {
+  ProtobufWkt::Value::KindCase kind = value.kind_case();
+
+  switch (kind) {
+  case ProtobufWkt::Value::kNullValue:
+    return lua_pushnil(state);
+
+  case ProtobufWkt::Value::kNumberValue:
+    return lua_pushnumber(state, value.number_value());
+
+  case ProtobufWkt::Value::kBoolValue:
+    return lua_pushboolean(state, value.bool_value());
+
+  case ProtobufWkt::Value::kStringValue: {
+    const auto string_value = value.string_value();
+    return lua_pushstring(state, string_value.c_str());
+  }
+
+  case ProtobufWkt::Value::kStructValue: {
+    return createTable(state, std::move(value.struct_value().fields()));
+  }
+
+  case ProtobufWkt::Value::kListValue: {
+    const auto list = value.list_value();
+    const int values_size = list.values_size();
+
+    lua_createtable(state, values_size, 0);
+    for (int i = 0; i < values_size; i++) {
+      setValue(state, std::move(list.values(i)));
+      lua_rawseti(state, -2, i + 1);
```

Comment:
Can you add a small comment on what this does? The Lua API is pretty "raw" and it's nice to not have to go look up what this magic is doing.